src/commands/env/env-set.mjs

```diff
@@ -88,15 +89,29 @@ const setInMongo = async ({ api, key, siteInfo, value }) => {
 
 /**
  * Updates the env for a site configured with Envelope with a new key/value pair
- * @returns {Promise<object>}
+ * @returns {Promise<object | boolean>}
  */
-const setInEnvelope = async ({ api, context, key, scope, siteInfo, value }) => {
+const setInEnvelope = async ({ api, context, key, scope, secret, siteInfo, value }) => {
   const accountId = siteInfo.account_slug
   const siteId = siteInfo.id
+
+  // secret values must specify deploy contexts. `all` or `dev` are not allowed
+  if (secret && value && (!context || context.includes('dev'))) {
+    error(
+      `To set a secret environment variable value, please specify a non-development context with the \`--context\` flag.`,
+    )
+    return false
+  }
+
   // fetch envelope env vars
   const envelopeVariables = await api.getEnvVars({ accountId, siteId })
   const contexts = context || ['all']
-  const scopes = scope || AVAILABLE_SCOPES
+  let scopes = scope || AVAILABLE_SCOPES
+
+  if (secret) {
+    // post_processing (aka post-processing) scope is not allowed with secrets
+    scopes = scopes.filter((sco) => !/post[-_]processing/.test(sco))
+  }
```

Comment:
We should probably error if the developer passed just `post-processing` right? Otherwise, it gets filtered out and looks empty and all of the empty fallbacks happen implicitly, right?

Reply:
Yep, if they explicitly pass in post-processing, it will be silently removed. Erroring out in this case sounds like a good solution 👍